### DISCUSSION TITLE
Potential bug with CombinerGenerator handling of boundaries/blocks

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#20823] [Q&A Meshing] [answered]
I've been trying to use combiner generator to combine meshes for two parts I need to 'glue' constraint together later. The two parts have the surfaces/boundaries where they will be glued defined in their own exodus files, but when I load them into the combiner, whichever mesh is given as the second input loses all of its boundaries in the new mesh generated. Basically, only blocks, boundaries and sets defined in the first mesh given as input "survive" the combining process. I can't seem to find any parameter that would affect this, so it seems to me like it could be a bug/oversight?

**Comment by Edward-Eth** [comment]
This example shows the sort of behaviour I am talking about. GeneratedMeshGenerator for top and bottom block generates side sets around each cube with the prefixes upper and lower respectively. When the combiner is given the input "inputs = 'Top_Block Bottom_Block' ", the side sets from the top block are still named, but the side sets from the lower block are now just numbered and have no names. If you swap the meshes in the combiner input then the top block side sets are just numbered and not named, and the lower block sets are fine. In the file I'm actually working on, it actually deleted the side sets from the second mesh, not just removed their names, so there's definitely some odd behaviour occuring.

```
[Mesh]
    [Top_Block]
        type = GeneratedMeshGenerator
        dim = 3
        nx = 10
        ny = 10
        nz = 10
        xmax = 2
        ymax = 2
        zmax = 2
        xmin = 0
        ymin = 0
        zmin = 1
        boundary_name_prefix = 'Upper'
        boundary_id_offset = 10
    []
    [Bottom_Block]
        type = GeneratedMeshGenerator
        dim = 3
        nx = 10
        ny = 10
        nz = 10
        xmax = 2
        ymax = 2
        zmax = 1
        boundary_name_prefix = 'Lower'
    []
    [Combine]
        type = CombinerGenerator
        inputs = 'Top_Block Bottom_Block'
        positions = '0 0 0 0.12 0.12 0'
    []
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

So the IDs get kept in the process, but the names of the sidesets are not transfered. If you can get by with the IDs for now, I ll try to make a patch (when I have time) to make sure the names of the sidesets in the combined mesh (eg the second one in the input here) are also transfered. 

Thanks for reporting this 
Guillaume

**Comment by Edward-Eth** [comment]
Cheers, there was also an issue with boundaries being lost during the combination process, but that was in a more complex file involving file meshes so I'll have a fiddle and see if I can narrow down what the issue there was.

**Reply by GiudGiud** [reply]
did the IDs overlap in each mesh?
There is some code in the mesh generator to prevent this being an issue, based on adding an offset, but maybe it s not working for you

**Reply by Edward-Eth** [reply]
That's a good question, they might well have done. I'll look into that and get back to you if it seems to be acting strangely.
